client/src/components/_common/AppIcon/AppIcon.jsx

```diff
@@ -4,14 +4,32 @@ import PropTypes from 'prop-types';
 import Icon from '_common/Icon';
 import './AppIcon.scss';
 
+const doesClassExist = (className) => {
+  // Check if the CSS class exists in the stylesheets
+  for (let sheet of document.styleSheets) {
+    try {
+      for (let rule of sheet.cssRules) {
+        if (
+          rule.selectorText &&
+          rule.selectorText.includes(`.${className}::before`)
+        ) {
+          return true;
+        }
+      }
+    } catch (e) {
+      // Handle cross-origin stylesheet errors
+      continue;
+    }
```

Comment:
I wonder if we can instead use css modules here (i.e. `import styles from "path/to/icon.fonts.css";`), and check to see if the className exists in the stylesheet, since we know all app icons exist in a single stylesheet.

Reply:
I was considering this, but will this work long-term since Wesley mentioned with the Cortal icon update everything in icon.fonts.css being replaced and moved to icon.css? Maybe I can check both files for classNames?

Reply:
Ah ok, in that case your suggestion of checking both files sounds good!

Reply:
Refactored code here. Now instead of checking all stylesheets on the current document page--it checks both icon.css and icon.fonts.css to determine if the className for the icon exists. Also fixed an issue with Jest/identity-obj-proxy with how it was handling the two css module imports. When in the testing environment we can expect the css file content to be of type object (bc of jest/identity-obj-proxy) so I wrote a conditional to handle this in the testing environment.  Whereas when we're just importing the two css files directly through the component and they're of type string--we can use the includes() function to check for class existence.